src/hammer-vlsi/hammer_vlsi/hammer_tool.py

```diff
@@ -452,57 +504,83 @@ def has_step(name: str) -> bool:
                 # Cajole the type checker into accepting that step is a HammerToolStep
                 step = cast(HammerToolStep, step)
                 check_hammer_step_function(step.func)
+        for pstep in persistent_steps:
+            if not isinstance(pstep, HammerToolHookAction):
+                raise ValueError("Element in List[HammerToolHookAction] is not a HammerToolHookAction")
```

Comment:
Should this be HammerToolStep?

Reply:
In the clarification, it needs to be `HammerToolHookAction` still because its `location` and `target_name` are accessed later